src/reporter/comments-reporter.ts

```diff
@@ -0,0 +1,194 @@
+import {
+  getGithubFilePath,
+  getScannerViolationType,
+  GithubComment,
+  GithubExistingComment,
+} from "../common";
+
+import core, { setFailed } from "@actions/core";
+import { Octokit } from "@octokit/action";
+import { context } from "@actions/github";
+import { BaseReporter } from "./reporter.types";
+import { ScannerViolation } from "../sfdxCli";
+
+const HIDDEN_COMMENT_PREFIX = "<!--sfdx-scanner-->";
```

Comment:
Is this necessary? Would it be possible to use the existing `| Engine | Category | Rule | Severity | Type | Message | File |` header as the canonical identifier of a generated comment?

Reply:
You're complete right, we could. The rational for a separate identifier is if the format changes, this will still work indefinitely, so we aren't tied to the formatting which may change over time

Reply:
gotcha - makes sense: good addition.